adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -162,8 +169,12 @@ private void handleEndpointConnection(final MqttEndpoint endpoint) {
             endpoint.reject(MqttConnectReturnCode.CONNECTION_REFUSED_SERVER_UNAVAILABLE);
 
         } else {
-
-            endpoint.publishHandler(message -> {
+            final AtomicBoolean clientConnected = new AtomicBoolean(false);
```

Comment:
The Vert.x 3.4.0 is already out. I exposed isConnected in the current master branch for the mqtt server. I think we have to wait for the 3.4.1 that could be not so far.